The game crashes when open "settings" from main menu [Linux, playtest-20160424]

Hi! When I'm clicking on "settings" button in the main game menu, the game immediately crashes with System.NullReferenceException . I believe it's a sound related problem if judge by the game output. Please find log files in the attachment. I tested it only on my Linux machine.

[exception.txt](https://github.com/OpenRA/OpenRA/files/233581/exception.txt)
[sound.txt](https://github.com/OpenRA/OpenRA/files/233582/sound.txt)

Sorry friends, I'm not really familiar with .Net. The condition
`if (IsWin32Atom(next) || next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)...`
gives me a compilation error:
`OpenRA.Platforms.Default/OpenAlSoundEngine.cs(59,8): error CS0103: The name `IsWin32Atom' does not exist in the current context`

--- a/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
+++ b/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
@@ -64,7 +64,7 @@ namespace OpenRA.Platforms.Default
 
 			do
 			{
-				var str = Marshal.PtrToStringAnsi(next);
+				var str = Marshal.PtrToStringAuto(next);
 				next += str.Length + 1;
 				devices.Add(str);
 			} while (Marshal.ReadByte(next) != 0);

--- a/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
+++ b/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
@@ -13,6 +13,7 @@ using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Runtime.InteropServices;
+using System.Text;
 using OpenAL;
 
 namespace OpenRA.Platforms.Default
@@ -65,7 +66,7 @@ namespace OpenRA.Platforms.Default
 			do
 			{
 				var str = Marshal.PtrToStringAuto(next);
-				next += str.Length + 1;
+				next += UnicodeEncoding.Default.GetByteCount(str) + 1;
 				devices.Add(str);
 			} while (Marshal.ReadByte(next) != 0);
 

--- a/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
+++ b/OpenRA.Platforms.Default/OpenAlSoundEngine.cs
@@ -56,18 +56,18 @@ namespace OpenRA.Platforms.Default
 
 			var devices = new List<string>();
 			var next = ALC10.alcGetString(IntPtr.Zero, type);
+			if (next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)
+			{
+				Log.Write("sound", "Failed to query OpenAL device list using {0}", label);
+				return new string[] { };
+			}
+
 			do
 			{
 				var str = Marshal.PtrToStringAnsi(next);
 				next += str.Length + 1;
 				devices.Add(str);
 			} while (Marshal.ReadByte(next) != 0);
-
-			if (AL10.alGetError() != AL10.AL_NO_ERROR)
-			{
-				Log.Write("sound", "Failed to query OpenAL device list using {0}", label);
-				return new string[] { };
-			}
 
 			return devices.ToArray();
 		}

